skimage/io/collection.py

```diff
@@ -126,81 +125,41 @@ class ImageCollection:
     Note that files are always returned in alphanumerical order. Also note
     that slicing returns a new ImageCollection, *not* a view into the data.
 
-    ImageCollection image loading can be customized through
-    `load_func`. For an ImageCollection ``ic``, ``ic[5]`` calls
-    ``load_func(load_pattern[5])`` to load that image.
-
-    For example, here is an ImageCollection that, for each video provided,
-    loads every second frame::
-
-      import imageio.v3 as iio3
-      import itertools
-
-      def vidread_step(f, step):
-          vid = iio3.imiter(f)
-          return list(itertools.islice(vid, None, None, step)
-
-      video_file = 'no_time_for_that_tiny.gif'
-      ic = ImageCollection(video_file, load_func=vidread_step, step=2)
-
-      ic  # is an ImageCollection object of length 1 because 1 video is provided
-
-      x = ic[0]
-      x[5]  # the 10th frame of the first video
-
-    Alternatively, if `load_func` is provided and `load_pattern` is a
-    sequence, an `ImageCollection` of corresponding length will be created,
-    and the individual images will be loaded by calling `load_func` with the
-    matching element of the `load_pattern` as its first argument. In this
-    case, the elements of the sequence do not need to be names of existing
-    files (or strings at all). For example, to create an `ImageCollection`
-    containing 500 images from a video::
-
-      class FrameReader:
-          def __init__ (self, f):
-              self.f = f
-          def __call__ (self, index):
-              return iio3.imread(self.f, index=index)
-
-      ic = ImageCollection(range(500), load_func=FrameReader('movie.mp4'))
-
-      ic  # is an ImageCollection object of length 500
-
-    Another use of `load_func` would be to convert all images to ``uint8``::
-
-      def imread_convert(f):
-          return imread(f).astype(np.uint8)
-
-      ic = ImageCollection('/tmp/*.png', load_func=imread_convert)
-
     Examples
     --------
-    >>> import imageio.v3 as iio3
-    >>> import skimage.io as io
-
-    # Where your images are located
-    >>> data_dir = os.path.join(os.path.dirname(__file__), '../data')
-
-    >>> coll = io.ImageCollection(data_dir + '/chess*.png')
-    >>> len(coll)
+    >>> import skimage as ski
+    >>> ski.data.download_all()  # Make sure all files are present in data_dir
+    >>> collection = ski.io.ImageCollection(ski.data.data_dir + '/chess*.png')
```

Comment:
ski.data.data_dir is not reliable any more after the refactoring of the fetcher.

Reply:
I don't follow? It's part `skimage.data`'s pulbic API, see 
https://github.com/scikit-image/scikit-image/blob/604d0776f882e2ba6c50a2b1fc254d718edb659a/skimage/data/__init__.pyi#L16

The actual value of the directory doesn't matter as long as it's populated which is taken care of by `ski.data.download_all()`...

Reply:
Thinking about this, we might add a comment that the filenames are not part of our public API, meaning `'/chess*.png'` and `"/palisades_of_vogt.tif"`?

I really would like to have examples that are covered by tests. We didn't notice for some time that the old examples in the notes were broken.